docs/pages/api/events.md

```diff
@@ -0,0 +1,87 @@
+---
+description: WebSocket events
+---
+
+# Events
+
+LiveCompositor is using WebSocket connection to send events to the connected clients. Supported events are listed below.
+
+### `VIDEO_INPUT_DELIVERED`
+
+```typescript
+type Event = {
+  type: "VIDEO_INPUT_DELIVERED";
+  input_id: string;
+}
+```
+
+The compositor received the input, and the first frames of that input are ready to be used. If you want to ensure that some inputs are ready before you send the [`start`](./routes.md#start) request, you can wait for those events for specific inputs.
+
+### `VIDEO_INPUT_PLAYING`
+
+```typescript
+type Event = {
+  type: "VIDEO_INPUT_PLAYING";
+  input_id: string;
+}
+```
+
+The compositor received the input and is using the first frame for rendering. This event will not be sent before the [`start`](./routes.md#start) request.
+
+This event is usually sent at the same time as `VIDEO_INPUT_DELIVERED` except for 2 cases:
+- Before [`start`](./routes.md#start) request.
+- If input has the `offset_ms` field defined.
+
+### `VIDEO_INPUT_EOS`
+
+```typescript
+type Event = {
+  type: "VIDEO_INPUT_EOS";
+  input_id: string;
+}
+```
+
+The input stream has ended and all the frames were already processed.
+
+It is possible that samples mixed using this input are still in the compositor pipeline, but after this event, you can safely unregister it.
```

Comment:
What do you mean here by "safely"?

Reply:
No frames/samples will be lost if you unregister it at this point.

Reply:
I wouldn't use the word "safely" in such a context. This might be ambiguous or imply that unregistering without receiving this message is "not safe". Just explain in the docs what you meant here

Reply:
> unregistering without receiving this message is "not safe".

It is unsafe, you are losing data :D. But yes I get your point, I don't think this would be confusing, but after thinking more about I think it might be better to drop that sentence entierlly.